packages/notification-service/src/firebase.ts

```diff
@@ -122,29 +104,44 @@ export function initializeDb() {
       rewardsSenders = snapshot?.val() ?? []
       console.debug('Rewards senders updated: ', rewardsSenders)
     },
-    (errorObject: any) => {
-      console.error('Rewards senders data read failed:', errorObject.code)
-    }
+    firebaseFetchError('Rewards senders')
   )
 
-  database.ref('/inviteRewardAddresses').on('value', (snapshot) => {
-    const addresses = snapshot?.val() ?? []
-    console.debug(`inviteRewardAddresses fetched: ${addresses}`)
-    if (addresses.length) {
-      inviteRewardsSenders = addresses
-    }
-  })
+  database.ref('/inviteRewardAddresses').on(
+    'value',
+    (snapshot) => {
+      const addresses = snapshot?.val() ?? []
+      console.debug(`inviteRewardAddresses fetched: ${addresses}`)
+      if (addresses.length) {
+        inviteRewardsSenders = addresses
+      }
+    },
+    firebaseFetchError('Invite rewards addresses')
+  )
 
   knownAddressesCache = new KnownAddressesCache()
   knownAddressesCache.startListening(database)
 }
 
-export function getTokenFromAddress(address: string) {
-  return registrations[address]?.fcmToken ?? null
+export async function getRegistration(address: string) {
+  let registration: Registration | undefined
+  await registrationsRef.once(
+    'value',
+    (snapshot) => {
+      registration = (snapshot?.val() || {})[address]
```

Comment:
This is still fetching the whole `registrations` node into memory and then getting just one value and ignoring the rest :/ 

What we should do instead is fetching the node `registrations/${address}` directly instead of fetching `registrations` and then getting the value for a single address in memory. To do this instead of the `registrationsRef` we can use `database.ref(`/registrations/${address}`)` or maybe there's a way to ask a sub-node from the `registrationsRef` directly.

Reply:
ooohhhhh i see what you mean! thanks for this, i'm on it :)